java/java.hints/test/unit/src/org/netbeans/modules/java/hints/infrastructure/TreeRuleTestBase.java

```diff
@@ -242,9 +242,11 @@ protected String performFixTest(String fileName, String code, int pos, String er
         String realCode = toCheckDocument.getText(0, toCheckDocument.getLength());
         
         //ignore whitespaces:
-        realCode = realCode.replaceAll("[ \t\n]+", " ");
+        realCode = realCode.replaceAll("[ \t\r\n]+", " ");
 
         if (golden != null) {
+	    //ignore CRLF/LF differences in golden:
+            golden = golden.replaceAll("[ \t\r\n]+", " ");
```

Comment:
this seems to go beyond line separators, `\\s+` might be the more direct approach to normalize whitespace?

Reply:
I didn't want to change the existing `realCode` replacement beyond handling `\r` in addition to `\n`. 
- I am unsure about the original test writers intention here. 
- My guess is that the code generation may add tabs and newlines as part of the re-formatting which is orthogonal to the test case. 
- On the other hand `\\s+` might go beyond to other whitespace characters also, including unicode sequences.

Further, my change to `golden` is only trying to establish a conformance to the pre-processing done on the `realCode`.

Reply:
> I am unsure about the original test writers intention here.

the comment above does say `//ignore whitespaces:`. It is used to match refactored code with golden code. It essentially tries to substract formatting from code snippets to make them comparable as Strings in test cases. I can't imagine tabs or other special whitespace codepoints would carry special meaning here.

As mentioned below golden files should all be checked out from git equally. the editor could choose to use localized line separators I suppose, but this would be the refactored code, not the golden files.

So this _might_ not be a good sign that this has to clean up line separators of golden code too after the other line separator change was made.

Reply:
Thank you very much for your reviews.

I'm happy to change to `\\s+` in the `realCode` preprocessing in this file, after testing it at my end.

Regarding the goldenfiles, please advise me, pursuant to the discussion in the other comment. 
For now I can test with just the *.gitattributes* change and leave the worrying about the future for the future. :)